strategizer/strategizer.js

```diff
@@ -102,6 +109,19 @@ class Strategizer {
       return true;
     });
 
+    let terminal = new Map();
```

Comment:
would it maybe be more readable, if this was refactored into a separate method? 
this method is already too long to my taste, so probably better not to make it any longer.. up to you, of course.